fact/path/__init__.py

```diff
@@ -0,0 +1,114 @@
+import os.path
+import re
+
+__all__ = [
+    'parse'
+    'run2tree_path',
+    'TemplateToPath',
+    'TreePath',
+]
+
+
+class TemplateToPath:
+    '''Turn a template like '/fac/raw/{Y}/{M}/{D}/{N}_{R}.fits.fz' into a path.
+    '''
+
+    def __init__(self, template):
+        self.template = template
+
+    def __call__(self, night, run=None, **kwargs):
+        d = dict(**kwargs)
+
+        if hasattr(night, 'fNight') and hasattr(night, 'fRunID'):
```

Comment:
what is this for? I do not know any python objects which have fNight and fRunID members in our current software stack.

Reply:
I do this ... *a lot*

    runinfo = pd.read_sql_table("RunInfo", fact.credentials.create_fact_db_foo_bar())
    for run in runinfo.itertuples():
        f = something.open(tree_path('base', 'suffix', run)

but you are right ... this has nothing to do here.

Reply:
Yeah, I do this quite often, too. But then I can't stand those `fSomethingStupid` names and almost always immediately rename them.

Reply:
Also, this could be

```
runinfo = pd.read_sql_table("RunInfo", fact.credentials.create_fact_db_foo_bar())
for run in runinfo.iterrows():
    f = something.open(tree_path('base', 'suffix', run.fNight, run.fRunID)
```
This is only slightly more to type and I do not particularly like an object called night to also have the runid. I think if want to have this functiionality it should use the `run` kwarg. As a night is only a night, a run might be identified by (night, runid)